giftcard/src/main/java/com/adyen/checkout/giftcard/GiftCardComponent.kt

```diff
@@ -37,98 +27,41 @@ private const val LAST_FOUR_LENGTH = 4
 class GiftCardComponent(
     savedStateHandle: SavedStateHandle,
     paymentMethodDelegate: GenericPaymentMethodDelegate,
+    private val giftCardDelegate: GiftCardDelegate,
     configuration: GiftCardConfiguration,
-    private val publicKeyRepository: PublicKeyRepository
 ) : BasePaymentComponent<GiftCardConfiguration, GiftCardInputData, GiftCardOutputData, GiftCardComponentState>(
     savedStateHandle,
     paymentMethodDelegate,
     configuration
 ) {
 
-    override fun getSupportedPaymentMethodTypes(): Array<String> = PAYMENT_METHOD_TYPES
-
-    private var publicKey: String? = null
-
     override var inputData: GiftCardInputData = GiftCardInputData()
 
     init {
-        fetchPublicKey()
-    }
+        giftCardDelegate.outputDataFlow
+            .filterNotNull()
+            .onEach { notifyOutputDataChanged(it) }
+            .launchIn(viewModelScope)
+
+        giftCardDelegate.componentStateFlow
+            .filterNotNull()
+            .onEach { notifyStateChanged(it) }
+            .launchIn(viewModelScope)
+
+        giftCardDelegate.exceptionFlow
+            .onEach { notifyException(it) }
+            .launchIn(viewModelScope)
 
-    private fun fetchPublicKey() {
         viewModelScope.launch {
-            publicKeyRepository.fetchPublicKey(
-                environment = configuration.environment,
-                clientKey = configuration.clientKey
-            ).fold(
-                onSuccess = { key ->
-                    publicKey = key
-                    createComponentState()
-                },
-                onFailure = { e ->
-                    notifyException(ComponentException("Unable to fetch publicKey.", e))
-                }
-            )
+            giftCardDelegate.fetchPublicKey()
```

Comment:
Is there a way to make the delegate `fetchPublicKey()` by itself on init? And If so, is there any advantage? 🤔

Reply:
Can't think of a good way to do this 🤔 We would have the pass a `CoroutineScope` somehow. We can't pass `viewModelScope` in the constructor, because the `ViewModel` is created after the delegate. So we would have to pass the scope after both are created, which is less elegant than this imo

Reply:
👍